migrations/sql/schema/schema_1.0.0.py

```diff
@@ -0,0 +1,383 @@
+    "pixelFlags_interpolatedCenter",
+    "pixelFlags_offimage",
+    "pixelFlags_saturated",
+    "pixelFlags_saturatedCenter",
+    "pixelFlags_suspect",
+    "pixelFlags_suspectCenter",
+    "pixelFlags_streak",
+    "pixelFlags_streakCenter",
+    "pixelFlags_injected",
+    "pixelFlags_injectedCenter",
+    "pixelFlags_injected_template",
+    "pixelFlags_injected_templateCenter",
+]
+
+
+def upgrade() -> None:
+    """Upgrade 'schema' tree from 0.1.1 to 1.0.0. This includes changes on
+    tickets DM-42435 and DM-41530.
+
+    Summary of changes on DM-42435:
+
+      - DiaSource and DiaForcedSource table schema changed to use
+        (visit: long, detector: short) columns instead of ccdVisitId:
+        - ccdVisitId was in PK of DiaForcedSource, need to rebuild PK
+        - IDX_DiaSource_ccdVisitId index has to be dropped and replaced by
+          IDX_DiaSource_visitDetector
+        - same for IDX_DiaForcedSource_visitDetector
+      - DetectorVisitProcessingSummary.detector column type changed from long
+        to short, but we never had that table in the actual schema yet, nothing
+        to do for this change.
+      - Instrument table was dropped, but we have not actually used that
+        either, nothing to do for it too.
+
+    Summary of changes on DM-41530:
+
+      - Dropped columns DiaObject.flags, DiaSource.flags, DiaForcedSource.flags
+        and SSObject.flags
+      - Added a number of boolean columns to DiaSource table
+    """
+    ctx = Context()
+
+    instrument_name = ctx.get_mig_option("instrument")
+    repo = ctx.get_mig_option("butler-repo")
+    if not instrument_name or not repo:
+        raise ValueError(
+            "This migration script requires butler repository abd instrument name. "
+            "Please use `--options butler-repo=REPO --options instrument=NAME` command line options."
+        )
+
+    _check_insert_ids(ctx)
+    _check_flags()
+
+    cdVisitIds = _query_ccd_visit_ids(ctx)
+    _LOG.info("Found %s distinct ccdVisitIds", len(cdVisitIds))
+
+    # Find the (un)packer for this instrument.
+    packer = _make_packer(repo, instrument_name)
+
+    # Make and fill a table mapping ccdVisit to visit+detector
+    _make_map_table(ctx, cdVisitIds, packer)
+
+    # Add and fill visit + detector columns
+    _add_visit_detector_columns(ctx)
+
+    # Drop old column and update indices.
+    _drop_ccd_visit(ctx)
+
+    # Drop mapping table.
+    _LOG.info("Dropping %s table", ID_MAP_TABLE_NAME)
+    op.drop_table(ID_MAP_TABLE_NAME, schema=ctx.schema)
+
+    # Add all new flag columns to DiaSource and fill them.
+    _add_flag_columns(ctx)
+
+    # Drop old flags columns from all tables.
+    _drop_flag_columns(ctx)
+
+    # Update metadata version.
+    version = revision.split("_")[-1]
+    ctx.apdb_meta.update_tree_version("schema", version)
+
+
+def downgrade() -> None:
+    """Downgrade would be too complicated for this migration."""
+    raise NotImplementedError()
```

Comment:
Main complication is that I'd need to write SQL that reconstructs `flags` from 32 boolean columns. It's not super-complicated and can be done, but I want to avoid it for now. If unavoidable I can spend a day for implementing/testing `downgrade` too.

Reply:
I doubt we'll ever want to downgrade this particular schema change, so I don't think it's worth the time. If we ever eventually need it, we can implement it then.